rules/group.go

```diff
@@ -251,13 +322,16 @@ func (g *Group) run(ctx context.Context) {
 				return
 			case <-tick.C:
 				missed := (time.Since(evalTimestamp) / g.interval) - 1
-				if missed > 0 {
-					g.metrics.IterationsMissed.WithLabelValues(GroupKey(g.file, g.name)).Add(float64(missed))
-					g.metrics.IterationsScheduled.WithLabelValues(GroupKey(g.file, g.name)).Add(float64(missed))
-				}
 				evalTimestamp = evalTimestamp.Add((missed + 1) * g.interval)
-
-				g.evalIterationFunc(ctx, g, evalTimestamp)
+				g.evaluateRuleGroup(ctx, evalTimestamp, missed)
+				if g.ShouldRestore() && g.Active() {
+					restoreStartTime := time.Now()
```

Comment:
Can we keep the comment
```
		// have updated the latest values, on which some alerts might depend.
                ...
```
This is useful information.

Reply:
Good point! Done